cmd/xl-storage-disk-id-check.go

```diff
@@ -177,13 +177,32 @@ func (e *lockedLastMinuteLatency) total() AccElem {
 	return e.lastMinuteLatency.getTotal()
 }
 
+var maxConcurrentOnce sync.Once
+
 func newXLStorageDiskIDCheck(storage *xlStorage, healthCheck bool) *xlStorageDiskIDCheck {
+	// diskMaxConcurrent represents maximum number of running concurrent
+	// operations for local and (incoming) remote disk operations.
+	//
+	// this value is a placeholder it is overridden via ENV for custom settings
+	// or this default value is used to pick the correct value HDDs v/s NVMe's
+	var diskMaxConcurrent = -1
+	maxConcurrentOnce.Do(func() {
+		s := env.Get("_MINIO_DRIVE_MAX_CONCURRENT", "")
+		if s == "" {
+			s = env.Get("_MINIO_DISK_MAX_CONCURRENT", "")
+		}
+		if s != "" {
+			diskMaxConcurrent, _ = strconv.Atoi(s)
+		}
+	})
+
 	if diskMaxConcurrent <= 0 {
 		diskMaxConcurrent = 512
 		if storage.rotational {
-			diskMaxConcurrent = 32
+			diskMaxConcurrent = int(storage.nrRequests)/2
```

Comment:
let's do if storage.nrRequests == 0, we can set it to 32 by default. I am not sure if nrRequets is always present in /sys/

Reply:
Oh yes I forgot to fix that.